src/Spout/Writer/XLSX/Internal/Worksheet.php

```diff
@@ -212,8 +212,10 @@ private function getCellXML($rowIndex, $cellNumber, $cellValue, $styleId)
         $columnIndex = CellHelper::getCellIndexFromColumnIndex($cellNumber);
         $cellXML = '<c r="' . $columnIndex . $rowIndex . '"';
         $cellXML .= ' s="' . $styleId . '"';
-
-        if (CellHelper::isNonEmptyString($cellValue)) {
+        
+        if (CellHelper::isFormulaString($cellValue)) {
+            $cellXML .= '><f>'.substr($cellValue,1).'</f><v>0</v></c>'; // seriously, that's it.
```

Comment:
I suspect that the `<v>0</v>` is not going to work across all softwares. Which software did you use to test your changes?

Reply:
Hi there, realise i left this a long time - i will try this.

S

Reply:
how is the code for .ods?

Reply:
I try do the change on code for .ods 
[ODS/Internal/Worksheet.php](https://github.com/box/spout/blob/master/src/Spout/Writer/ODS/Internal/Worksheet.php)
```php
        if (CellHelper::isFormulaString($cellValue)) {
            $data .= ' table:formula="'.$cellValue.'" office:value-type="float" office:value="' . 0 . '" calcext:value-type="float">';
            $data .= '<text:p>' . 0 . '</text:p>';
            $data .= '</table:table-cell>';
        }else if (CellHelper::isNonEmptyString($cellValue)) {
            $data .= ' office:value-type="string" calcext:value-type="string">';
            $cellValueLines = explode("\n", $cellValue);
            foreach ($cellValueLines as $cellValueLine) {
                $data .= '<text:p>' . $this->stringsEscaper->escape($cellValueLine) . '</text:p>';
            }
            $data .= '</table:table-cell>';    
        } else if (CellHelper::isBoolean($cellValue)) {...}
```
too add the same change the [CellHelper.php](https://github.com/box/spout/blob/master/src/Spout/Writer/Common/Helper/CellHelper.php) like .xlsx

```php
 public static function isFormulaString($value)
    {
        return (strpos($value,'=') === 0);
    }
```
but now have this problem
not reconize the form in ods like excel. i neet to open document "make change at formule" and save to func 
![image](https://user-images.githubusercontent.com/25001142/38939990-c7e0eee8-42ee-11e8-9274-1ffb251ef41e.png)
![image](https://user-images.githubusercontent.com/25001142/38940034-e0892b90-42ee-11e8-9335-f95f15f84ba4.png)

this is the code .fods
```php
 <table:table-row table:style-name="ro1">
     <table:table-cell office:value-type="float" office:value="1" calcext:value-type="float">
      <text:p>1</text:p>
     </table:table-cell>
     <table:table-cell office:value-type="float" office:value="9" calcext:value-type="float">
      <text:p>9</text:p>
     </table:table-cell>
     <table:table-cell office:value-type="float" office:value="3" calcext:value-type="float">
      <text:p>3</text:p>
     </table:table-cell>
     <table:table-cell office:value-type="float" office:value="7" calcext:value-type="float">
      <text:p>7</text:p>
     </table:table-cell>
     <table:table-cell table:formula="of:=[.A1]+[.B1]+[.C1]+[.D1]" office:value-type="float" office:value="20" calcext:value-type="float">
      <text:p>20</text:p>
     </table:table-cell>
    </table:table-row>
```

if any can help me i give thanks

Reply:
Well, you can see that formulas don't work the same:
```
<table:table-cell table:formula="of:=[.A1]+[.B1]+[.C1]+[.D1]" office:value-type="float" office:value="20" calcext:value-type="float">
      <text:p>20</text:p>
```
This would indicate a formula: `table:formula="of:=[.A1]+[.B1]+[.C1]+[.D1]"`, so your `isFormulaString` implementation is incorrect. Setting the value to 0 may not work in ODS. This is a trick in XLSX because softwares do recompute the value when you open the file, but for ODS maybe softwares use the given value directly. In this case, I don't have a good solution for you.